python-avd/pyavd/_eos_designs/structured_config/base/snmp_server.py

```diff
@@ -60,26 +60,73 @@ def snmp_server(self: AvdStructuredConfigBaseProtocol) -> None:
+        if not snmp_settings._get("compute_local_engineid_rfc3411", None):
+            # This generation algorithm is not RFC3411 compliant, but needs to be maintained for backward compatibility
+            # If the mgmt_ip is not set, this will hash hostname+"None"
+            return sha1(f"{self.shared_utils.hostname}{self.shared_utils.node_config.mgmt_ip}".encode()).hexdigest()  # NOSONAR # noqa: S324
+        if self.shared_utils.node_config.mgmt_ip is None:
+            msg = "hostname_oob_ip engine id calculation requires mgmt_ip to be set when set being RFC3411 compliant."
+            raise AristaAvdInvalidInputsError(msg)
+        return self._snmp_engine_id_hash(self.shared_utils.node_config.mgmt_ip)
+
     def _snmp_engine_ids(self: AvdStructuredConfigBaseProtocol, snmp_settings: EosDesigns.SnmpSettings) -> None:
         """Set dict of engine ids if "snmp_settings.compute_local_engineid" is True."""
         if not snmp_settings.compute_local_engineid:
             return
 
-        compute_source = snmp_settings.compute_local_engineid_source
-        if compute_source == "hostname_and_ip":
-            # Accepting SonarLint issue: The weak sha1 is not used for encryption. Just to create a unique engine id.
-            local_engine_id = sha1(f"{self.shared_utils.hostname}{self.shared_utils.node_config.mgmt_ip}".encode()).hexdigest()  # NOSONAR # noqa: S324
-
-        elif compute_source == "system_mac":
-            if self.shared_utils.system_mac_address is None:
-                msg = "default_engine_id_from_system_mac: true requires system_mac_address to be set."
-                raise AristaAvdInvalidInputsError(msg)
-            # the default engine id on switches is derived as per the following formula
-            local_engine_id = f"f5717f{str(self.shared_utils.system_mac_address).replace(':', '').lower()}00"
-        else:
-            # Unknown mode
-            msg = f"'{compute_source}' is not a valid value to compute the engine ID, accepted values are 'hostname_and_ip' and 'system_mac'"
-            raise AristaAvdError(msg)
+        match snmp_settings.compute_local_engineid_source.lower():
+            case "hostname_and_ip":
+                # This is the default value in AVD 5.x
+                # This should be changed to self.shared_utils.default_mgmt_method when mgmt_ip is enforced
+                # when the default oob method is present
+                if self.inputs.default_mgmt_method and self.inputs.default_mgmt_method == "inband":
+                    local_engine_id = self._snmp_engine_ids_inband()
+                else:
+                    # For backward compatibility, the default should be oob
+                    local_engine_id = self._snmp_engine_ids_oob(snmp_settings)
+
+            case "system_mac":
+                if self.shared_utils.system_mac_address is None:
+                    msg = "default_engine_id_from_system_mac: true requires system_mac_address to be set."
```

Comment:
cosmetic but lets fix it here (will need to update a negative unit test)
```suggestion
                    msg = "'default_engine_id_from_system_mac: true' requires 'system_mac_address' to be set."
```